### DISCUSSION TITLE
Images have stopped displaying

### DISCUSSION THREAD

**Discussion by ruapehujames51** [netbox-community/netbox#14115] [Help Wanted!]
HI new here to Netbox and really like it want to use it for documenting our sites and have been loving it but I have an issue with images no longer displaying.
Nothing appears in the images section of locations where the images were uploaded.
When I go to image attachment section under customisation I get this error.
![image](https://github.com/netbox-community/netbox/assets/148921750/7cbe36ac-f3c7-4d73-9dd2-8a88b3fa19a8)
Any help on this error is appreciated.

James

**Comment by bmp521** [comment]
I just noticed that I'm having the same issue and same error. I'm running 3.5.8 and was preparing to update to the latest version. The last command I ran was "wc -l" by accident instead of "ls -1 | wc -l" to get a total count of images to compare against the Netbox Image Attachment list. I didn't expect "wc -l" to modify anything. Could this cause the issue?

**Reply by bmp521** [reply]
I ended up restoring from a backup the other day since there were only minimal changes. The images worked for a few days but I'm back to the error again.

**Comment by candlerb** [comment]
What *exact* URL was in the browser's URL bar at the time the exception occurred? That is, were you trying to look at the listing (`/extras/image-attachments/`), or view one attachment (`/extras/image-attachments/N/`), or you were in the edit view (`/extras/image-attachments/N/edit/`)? Or did it only happen when you were saving a change from the edit view?

I suggest that when the error occurs again, get a full backtrace (set ADMINS and EMAIL in configuration.py and it will be E-mailed to you automatically; or wait until the problem occurs and set DEBUG=True and you'll get it in the web interface)

It's quite possibly a bug. I see some references to "parent" in `netbox/extras/views.py` in the edit and delete views for image attachments:

```
@register_model_view(ImageAttachment, 'edit')
class ImageAttachmentEditView(generic.ObjectEditView):
...

    def alter_object(self, instance, request, args, kwargs):
        if not instance.pk:
            # Assign the parent object based on URL kwargs
            content_type = get_object_or_404(ContentType, pk=request.GET.get('content_type'))
            instance.parent = get_object_or_404(content_type.model_class(), pk=request.GET.get('object_id'
))
        return instance

    def get_return_url(self, request, obj=None):
        return obj.parent.get_absolute_url() if obj else super().get_return_url(request)
```

And in the model:

```
    def to_objectchange(self, action):
        objectchange = super().to_objectchange(action)
        objectchange.related_object = self.parent
        return objectchange
```

These don't seem to have changed from 3.5.8 to 3.6.4

Also read through the later release notes, although I don't see any bugfixes for image attachments after 3.5.3, so it may not be fixed in 3.6.x

**Comment by bmp521** [comment]
Brian,

Thank you for the email. I'm using the following URL:
https://netbox-server/extras/image-attachments from the left menu pane,
Customization, Image Attachments. Also, for locations or racks, the images
are no longer listed. The images are in fact in the media directory. I do
have some concern regarding the mismatch in permissions as shown below but
that may be irrelevant to this issue.

I was using 3.5.8 when I noticed the issue. I tried upgrading to 3.6.4,
which had no effect. I then restored a VM backup to 3.5.8 to the day before
I noticed the issue. The images worked properly at that time. Then, I
upgraded the restored VM to 3.6.4 and the images were still working
properly for a day or two. I'm not sure if the issue was caused by me
attempting to upload a new image but that's the only change I can think of
that could be related.

I have enabled debug and copied the results to a text file. Is it safe to
attach the full log to an email to you or could I encrypt it with your
public key?  Here are the beginning debug details in yellow at the top of
the log if this helps.

FieldError at /extras/image-attachments/

Field 'parent' does not generate an automatic reverse relation and
therefore cannot be used for reverse querying. If it is a
GenericForeignKey, consider adding a GenericRelation.

Request Method: GET
Request URL: https://cone-svnetbox/extras/image-attachments/
Django Version: 4.2.6
Exception Type: FieldError
Exception Value:

Field 'parent' does not generate an automatic reverse relation and
therefore cannot be used for reverse querying. If it is a
GenericForeignKey, consider adding a GenericRelation.

Exception Location:
/opt/netbox-3.6.4/venv/lib/python3.10/site-packages/django/db/models/sql/query.py,
line 1700, in names_to_path
Raised during: extras.views.ImageAttachmentListView
Python Executable: /opt/netbox-3.6.4/venv/bin/python3
Python Version: 3.10.12
Python Path:

['/opt/netbox/netbox',
 '/opt/netbox-3.6.4',
 '/opt/netbox-3.6.4/venv/bin',
 '/usr/lib/python310.zip',
 '/usr/lib/python3.10',
 '/usr/lib/python3.10/lib-dynload',
 '/opt/netbox-3.6.4/venv/lib/python3.10/site-packages']

Server time: Fri, 03 Nov 2023 16:27:59 +0000


Here is an example of the mismatch of the image file permissions between root and netbox that I
mentioned above.

/opt/netbox/netbox/media/image-attachments$ ls -l
total 194932
-rw-r--r-- 1 root   root   2736337 Feb 17  2023 device_970_UPS_in_Rack_3.png
-rw-r--r-- 1 root   root   2736337 Feb 17  2023 device_971_UPS_in_Rack_3.png
-rw-r--r-- 1 root   root   3482887 Feb 17  2023
location_107_MDF_Rack_1_Top.png
-rw-r--r-- 1 root   root   2804481 Feb 17  2023
location_107_Rack_1_Bottom.png
-rw-r--r-- 1 root   root   2804481 Feb 17  2023
location_107_Rack_2_Bottom.png
-rw-r--r-- 1 root   root   3200480 Feb 17  2023 location_107_Rack_2_Top.png
-rw-r--r-- 1 root   root   2736337 Feb 17  2023
location_107_Rack_3_Bottom.png
-rw-r--r-- 1 root   root   2724260 Feb 17  2023 location_107_Rack_3_Top.png
-rw-r--r-- 1 root   root    750323 Feb 17  2023
location_10_IDF_Rack_1.png
-rw-r--r-- 1 netbox netbox  179548 Jul 25 19:46
location_114_MDF.jpg
-rw-r--r-- 1 netbox netbox 1555327 May 17 20:19
location_117_B-Front.jpg
-rw-r--r-- 1 netbox netbox 1153418 May 17 20:19
location_117_B-Rear.jpg



Thank you for your support,
Patrick McGlynn

**Reply by candlerb** [reply]
The backtrace, which is a list of python line numbers and source lines, is the most important part. It's safe to show, please just post it into this thread. Put three backticks (` ``` `) on a line of their own, before and after.

Imagine file permissions are unlikely to be anything to do with this; it's a problem with the code which interacts with the database.

You can fix the permissions using `chown --recursive` and making sure that netbox is running as the netbox user, not as root.  Instructions for this are in the installation instructions [here](https://docs.netbox.dev/en/stable/installation/3-netbox/#create-the-netbox-system-user).

**Reply by bmp521** [reply]
```
Traceback Switch to copy-and-paste view
/opt/netbox-3.6.4/venv/lib/python3.10/site-packages/django/core/handlers/exception.py, line 55, in inner
                response = get_response(request) …
Local vars
/opt/netbox-3.6.4/venv/lib/python3.10/site-packages/django/core/handlers/base.py, line 197, in _get_response
                response = wrapped_callback(request, *callback_args, **callback_kwargs) …
Local vars
/opt/netbox-3.6.4/venv/lib/python3.10/site-packages/django/views/generic/base.py, line 104, in view
            return self.dispatch(request, *args, **kwargs) …
Local vars
/opt/netbox/netbox/netbox/views/generic/base.py, line 77, in dispatch
        return super().dispatch(request, *args, **kwargs) …
Local vars
/opt/netbox/netbox/utilities/views.py, line 99, in dispatch
        return super().dispatch(request, *args, **kwargs) …
Local vars
/opt/netbox-3.6.4/venv/lib/python3.10/site-packages/django/views/generic/base.py, line 143, in dispatch
        return handler(request, *args, **kwargs) …
Local vars
/opt/netbox/netbox/netbox/views/generic/bulk_views.py, line 163, in get
        table = self.get_table(self.queryset, request, has_bulk_actions) …
Local vars
/opt/netbox/netbox/netbox/views/generic/mixins.py, line 47, in get_table
        table.configure(request) …
Local vars
/opt/netbox/netbox/netbox/tables/tables.py, line 156, in configure
                self.order_by = ordering …
Local vars
/opt/netbox-3.6.4/venv/lib/python3.10/site-packages/django_tables2/tables.py, line 542, in order_by
        self.data.order_by(self._order_by) …
Local vars
/opt/netbox-3.6.4/venv/lib/python3.10/site-packages/django_tables2/data.py, line 224, in order_by
            self.data = self.data.order_by(*order_by_accessors) …
Local vars
/opt/netbox-3.6.4/venv/lib/python3.10/site-packages/django/db/models/query.py, line 1659, in order_by
        obj.query.add_ordering(*field_names) …
Local vars
/opt/netbox-3.6.4/venv/lib/python3.10/site-packages/django/db/models/sql/query.py, line 2221, in add_ordering
                self.names_to_path(item.split(LOOKUP_SEP), self.model._meta) …
Local vars
/opt/netbox-3.6.4/venv/lib/python3.10/site-packages/django/db/models/sql/query.py, line 1700, in names_to_path
                    raise FieldError( …
Local vars
```

**Reply by bmp521** [reply]
I agree with you on the file permissions and will look at updating them. Thank you.

**Reply by candlerb** [reply]
Are you sure that's the  complete backtrace? There's not another set of lines, with "While handling this exception ..."?

**Reply by bmp521** [reply]
No, "while handling this exception" is not present in the debug file.

**Reply by bmp521** [reply]
Is there any additional logs or details I can provide that would help? Should I consider rolling back to an earlier release?

**Comment by bmp521** [comment]
Also, I followed the instructions to update the file permissions within the media directory.

**Comment by bmp521** [comment]
Please see Issue #14247 for the solution. Thank you all for your assistance!
